src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -528,7 +528,13 @@ public function morphToMany($related, $name, $table = null, $foreignPivotKey = n
         // Now we're ready to create a new query builder for this related model and
         // the relationship instances for this relation. This relations will set
         // appropriate query constraints then entirely manages the hydrations.
-        $table = $table ?: Str::plural($name);
+        if (! $table) {
+            $words = preg_split('/(_)/u', $name, -1, PREG_SPLIT_DELIM_CAPTURE);
+
+            $lastWord = array_pop($words);
+
+            $table = implode('', $words).Str::plural($lastWord);
```

Comment:
`PREG_SPLIT_DELIM_CAPTURE` makes it also capture the delimiter (in this case the underscore), so it can be imploded without a delimiter to rebuild the original string. I copied this code from the `Str::pluralStudly` method. Doing it this way is necessary when exploding studly case strings, it isn't really necessary to do it this way for snake case strings.

It could be changed to more familiar php like this:
```php
$words = explode('_', $name);

$words[count($words) - 1] = Str::plural($words[count($words) - 1]);

$table = implode('_', $words);
```